src/hb-subset-plan.cc

```diff
@@ -96,7 +125,13 @@ _gsub_closure_glyphs_lookups_features (hb_face_t *face,
                                  nullptr,
                                  nullptr,
                                  &feature_indices);
-  gsub->prune_features (gsub_lookups, &feature_indices);
+
+  hb_map_t duplicate_feature_map;
+  gsub->prune_features (gsub_lookups, &feature_indices, &duplicate_feature_map);
+  _remap_feature_indexes (&feature_indices, &duplicate_feature_map, gsub_features_org);
```

Comment:
I think this step (remap_feature_indexes) can be eliminated. If you change the duplicate feature finder method to put all retained feature indices into the duplicate feature map (mapped to themselves if they are not duplicates) then duplicate_feature_map can be used in place of gsub_features_org. ie. the subset plan would retain a reference to duplicate_feature_map instead of gsub_features_org. and collect_features would use that instead.

I found the name 'gsub_features_org' to not be intuitive as to what it's purpose was, using duplicate_feature_map instead throughout makes it more clear.